### PR TITLE
Ride dialog

### PR DESCRIPTION
**Pull request by jonzee**

### CODE REVIEW

File: src/sass/Detail.scss [hunk 1]
```diff
@@ -1,91 +1,27 @@
...
   min-width: 300px;
+  min-height: 650px;
   padding-top: 112px;
   display: flex;
   flex-direction: row;
   justify-content: space-between;
   .image {
     margin-right: 100px;
-    height: 420px;
+    height: $detail-image-height;
```
Comment (michalmikolajczyk) [inline src/sass/Detail.scss]: cool

File: src/ts/Ride/FormVG.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    const price = 55;
+    const start = moment().toDate();
+    wrapper.instance()['startDateChange'](undefined, start);
+    _.forEach(days, (day) => {
+      const end = day > 0 ? moment().add(day, 'days').toDate() : moment().subtract(day, 'days').toDate();
+      wrapper.instance()['endDateChange'](undefined, end);
+      expect(instance.getTotal()).to.be.equal(((Math.abs(day) + 1) * price));
+    });
+  });
+});
```
Comment (michalmikolajczyk) [inline src/ts/Ride/FormVG.spec.tsx]: <3

File: src/ts/Ride/FormWrap.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+export default class FormWrap extends React.Component<any, any> {
+  requestId: number;
+  state = { form: '' }
+
+  componentDidMount() {
+    this.handleScroll();
+  }
+
+  componentWillUnmount() {
+    window.cancelAnimationFrame(this.requestId);
```
Comment (michalmikolajczyk) [inline src/ts/Ride/FormWrap.tsx]: does this work ok with tests and server-side rendering?

File: test-setup.js [hunk 4]
```diff
@@ -14,6 +14,7 @@ global.window.fetch = require('whatwg-fetch');
 
 global.window.requestAnimationFrame = callback => {
   global.window.setTimeout(callback, 1000 / 60);
+  return 2;
```
Comment (michalmikolajczyk) [inline test-setup.js]: ?

File: src/ts/Ride/FormWrap.tsx (hunk 3 above)
Comment (jonzee) [inline src/ts/Ride/FormWrap.tsx]: Yep, for tests I added in `test-setup.js` `requestAnimationFrame` that returns 2 (`requestId`). On server-side rendering methods like `componentDidMount` or `componentWillUnmout` aren't invoked.

File: test-setup.js (hunk 4 above)
Comment (jonzee) [inline test-setup.js]: `requestAnimationFrame` should return `request id` (mockup for tests)

File: src/ts/Ride/FormPure.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    equipment,
+    startDate,
+    endDateChange,
+    equipmentChange,
+    startDateChange,
+  } = props
+
+  const localPrice = fx(price).to(i('USD')).toFixed(2);
+  const localTotal = fx(total).to(i('USD')).toFixed(2);
+  return (
```
Comment (michalmikolajczyk) [inline src/ts/Ride/FormPure.tsx]: break line before new block

File: src/ts/Ride/FormVG.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+import { shallow } from 'enzyme';
+import * as sinon from 'sinon';
+require('sinon-stub-promise')(sinon);
+import * as sinonChai from 'sinon-chai';
+chai.use(sinonChai);
+const expect = chai.expect;
+import { formsyContext } from '../helpers/test-theme';
+import * as moment from 'moment';
+import * as _ from 'lodash';
+
```
Comment (michalmikolajczyk) [inline src/ts/Ride/FormVG.spec.tsx]: rm nl

**Pull request merged**